dbsnp-importer/src/main/java/uk/ac/ebi/eva/dbsnpimporter/models/SubSnpGenotype.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2017 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.ac.ebi.eva.dbsnpimporter.models;
+
+
+import uk.ac.ebi.eva.dbsnpimporter.io.readers.SubSnpGenotypeRowMapper;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Representation for Genotypes associated with a given SubSNP
+ */
+public class SubSnpGenotype {
+
+    private int batchId;
+
+    private String studyId;
+
+    private long ssId;
+
+    private List<String> genotypes;
+
+    /**
+     * @param batchId           Unique Batch identifier (maps to "fid" attribute in
+     *                          files sub-document under the variants document in Mongo)
+     * @param studyId        Unique study identifier (maps to "sid" attribute in
+     *                          files sub-document under the variants document in Mongo)
+     * @param subSnpId          Unique SS ID identifier
+     * @param genotypes         Genotypes associated with the SS
+     */
+    public SubSnpGenotype(int batchId, String studyId, long subSnpId, String genotypes) {
+        this.batchId = batchId;
+        this.studyId = studyId;
+        this.ssId = subSnpId;
+        if (genotypes == null || genotypes.trim().equals("")) {
+            this.genotypes = new ArrayList<>();
+        }
```

Comment:
Just one last correction. The if - else structure should be:

if (whatever) {
........
} else {
......
}
 
The else has to be in the same line than the latest "}"